lib/dao.js

```diff
@@ -468,6 +468,18 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   var Model = this;
   var connector = Model.getConnector();
 
+  if (Array.isArray(data)) {
+    var isBulkCreate = data.every(function(item){
```

Comment:
What does this mean? Are you trying to say "if every item does not have an id value, then call the bulk create function"?